### ISSUE TITLE
function-url-data-uris

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#996]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Disable usage inline media in `url` function.

> Which rule, if any, is this issue related to?

`function-url-no-inline`

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  background-image: url(data:image/gif;base64,R0lGODlhDAANANUAAPv7+/r6+vn5+ff39/b29vX19fPz8/Dw8O/v7+7u7u3t7ezs7Onp6ejo6Ofn5+bm5uTk5OPj4+Hh4eDg4N/f397e3tzc3Nvb29ra2tnZ2dbW1tXV1dPT09LS0tHR0c3NzcjIyMfHx8bGxsXFxby8vLe3t7W1tbOzs62traurq6mpqaioqKenp6ampqWlpaOjo6KioqGhoaCgoJ+fn5qampmZmf///wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAUUADYALAAAAAAMAA0AAAZxQJtwyGYrVaLEQswGYWAOAygxWELxaFwCVUWi/bRyZaJM5nUezjInkcjHi8U2rVNhmJY++QZDg0NBgTExEPhxJ+NCogDY6PjiMqGisLCJeYDTIaNikmCgOhAwgnKUICKCsdBgYdMSgCRBgogSgZQ0EAOw==);
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "function-url-no-inline": true
  }
}
```

> Which version of stylelint are you using?

`5.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

no matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no matter

> What did you expect to happen?

Throw errors.

> What actually happened (e.g. what warnings or errors you are getting)?

Nothing

**Comment by jeddy3** [comment]
You're on a roll with your new rule suggestions! :)

I assume the use case is that you'd probably want to use tooling to inline your assets, rather than manually maintaining the data URIs. So, this feels like a good rule to me.

I think there's value in allowing the opposite to be enforced too i.e. only allowing data URIs. For example, if you wanted to lint the output of your tooling to check all the `url` functions were correctly processed.

How about `function-url-data-uris: "always"|"never"`?

Seems like a pretty simple rule to implement using `postcss-value-parser`? For each `function` check that the value either starts within `"data:"` or not?

**Comment by evilebottnawi** [comment]
@jeddy3 look good to me.

**Comment by evilebottnawi** [comment]
@jeddy3 wip

**Comment by jeddy3** [comment]
Thanks again @evilebottnawi
